packages/sdk/src/rpc/rpc.ts

```diff
@@ -0,0 +1,212 @@
+   * @async
+   * @returns {string} Address of native token
+   */
+  async queryNativeToken(): Promise<string> {
+    return await this.query.query_native_token();
+  }
+
+  /**
+   * Query public key
+   * Return string of public key if it has been revealed on chain, otherwise, return null
+   * @async
+   * @param {string} address - Address to query
+   * @returns {string|null} String of public key if found
+   */
+  async queryPublicKey(address: string): Promise<string | undefined> {
+    const pk = await this.query.query_public_key(address);
+    return pk;
+  }
+
+  /**
+   * Query all validator addresses
+   * @async
+   * @returns {string[]} Array of all validator addresses
+   */
+  async queryAllValidators(): Promise<string[]> {
+    return await this.query.query_all_validator_addresses();
+  }
+
+  /**
+   * Query Proposals
+   * @async
+   * @returns {Proposal[]} List of the proposals
+   */
+  async queryProposals(): Promise<Proposal[]> {
+    const serializedProposals = await this.query.query_proposals();
+    const { proposals } = deserialize(serializedProposals, Proposals);
+    return proposals;
+  }
+
+  /**
+   * Query total delegations
+   * @async
+   * @param {string[]} owners - Array of owner addresses
+   * @param {bigint} [epoch] - delegations at epoch
+   * @returns {Promise<DelegationTotals>} Promise resolving to total delegations
+   */
+  async queryTotalDelegations(
+    owners: string[],
+    epoch?: bigint
+  ): Promise<DelegationTotals> {
+    return await this.query.get_total_delegations(owners, epoch);
+  }
+
+  /**
+   * Query delegators votes
+   * @async
+   * @param {bigint} proposalId - ID of the proposal
+   * @returns {Promise<DelegatorsVotes>} Promise resolving to delegators votes
+   */
+  async queryDelegatorsVotes(proposalId: bigint): Promise<DelegatorsVotes> {
+    return await this.query.delegators_votes(proposalId);
+  }
+
+  /**
+   * Query staking totals by owner addresses
+   * @async
+   * @param {string[]} owners - Array of owner addresses
+   * @returns {Promise<StakingTotals[]>} Promise resolving to staking totals
+   */
+  async queryStakingTotals(owners: string[]): Promise<StakingTotals[]> {
+    const stakingAmounts = await this.query.query_my_validators(owners);
+    return stakingAmounts.map(
+      ([
+        owner,
+        validator,
+        bonds,
+        unbonds,
+        withdrawable,
+      ]: StakingTotalsResponse) => {
+        return {
+          owner,
+          validator,
+          bonds,
+          unbonds,
+          withdrawable,
+        };
+      }
+    );
+  }
+
+  /**
+   * Query bond and unbond details by owner addresses
+   * @async
+   * @param {string[]} owners - Array of owner addresses
+   * @returns {Promise<StakingPositions>} Promise resolving to staking positions
+   */
+  async queryStakingPositions(owners: string[]): Promise<StakingPositions> {
+    const [bonds, unbonds] = await this.query.query_staking_positions(owners);
```

Comment:
Related to the comment on the `StakingPositions` type, it looks like the type annotations need to be added earlier for TypeScript to catch the problem:
```ts
const [bonds, unbonds]: [BondsResponse[], UnbondsResponse[]] =
  await this.query.query_staking_positions(owners);
```

Reply:
I think we will catch more of those in the 2nd PR as we will replace shared with sdk in the extension :)